[Bug]: aws_secretsmanager_secret_version produces "inconsistent final plan" (Update → DeleteThenCreate) after 6.45.0 when secret_string depends on a replaced upstream  resource

### Terraform and AWS Provider Version

```console
- OpenTofu: OpenTofu v1.12.0
  - AWS provider: 6.45.0 (regression) vs 6.44.0 (works)
  - OS: ubuntu-latest (GitHub Actions)
```

### Affected Resource(s) or Data Source(s)

  - `aws_secretsmanager_secret_version`

### Expected Behavior

  aws_secretsmanager_secret_version plans and applies an Update (matches 6.44.0
  behavior). The has_secret_string_wo attribute should stay false/null when
  secret_string_wo is unset.


### Actual Behavior

  Plan shows Update. During apply, after upstream values become known, provider raises:

  Error: Provider produced inconsistent final plan

  When expanding the plan for
  module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["X"]
  to include new values learned so far during apply, provider
  "registry.opentofu.org/hashicorp/aws" changed the planned action from
  Update to DeleteThenCreate.

  …followed by arn, id, version_id, version_stages, has_secret_string_wo,
  and secret_binary all reported as inconsistent.

  After upgrading from `hashicorp/aws` 6.44.0 → 6.45.0, `tofu apply` fails with
  "Provider produced inconsistent final plan" on every `aws_secretsmanager_secret_version`
  whose `secret_string` is computed from an upstream resource being replaced in the same
  apply. No other changes to config, OpenTofu, or workflow.

  Our resources use plain `secret_string` only — we do not set `secret_string_wo` anywhere.
  The errors specifically reference the new `has_secret_string_wo` attribute:

  > `has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool)`

  and report the provider flipping the planned action mid-apply:

  > changed the planned action from Update to DeleteThenCreate

  This appears to be a regression introduced by the 6.45.0 enhancement
  ["Allow switching from `secret_string` to `secret_string_wo` without re-creating
  the resource"](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v6.45.0),
  which is likely the upstream fix for #41635. The new logic seems to engage even when
  `secret_string_wo` is never used, in any apply where `secret_string` resolves from
  `(known after apply)` values.


  Workaround -> Pin version = "= 6.44.0".

### Relevant Error/Panic Output

```console

  Full Log:
---
Plan: 9 to add, 8 to change, 9 to destroy.
module.wacnet.aws_eip_association.wg_association: Destroying... [id=eipassoc-09908d6c62d6fccde]
module.wacnet.aws_eip_association.wg_association: Destruction complete after 1s
module.wacnet.aws_instance.wg_gateway: Destroying... [id=i-04551508069f91011]
module.wacnet.aws_instance.wg_gateway: Still destroying... [id=i-04551508069f91011, 10s elapsed]
module.wacnet.aws_instance.wg_gateway: Still destroying... [id=i-04551508069f91011, 20s elapsed]
module.wacnet.aws_instance.wg_gateway: Destruction complete after 30s
module.wacnet.wireguard_asymmetric_key.peers["mlibera"]: Destroying... [id=xpKf/IhGkuk5UKVHK9wb2V4pffii/9W861dcaDUNmXU=]
module.wacnet.wireguard_asymmetric_key.peers["mschroeder"]: Destroying... [id=IFuGued1lglxFle+TyaLrPOZtNQRFzhgRE6any9D51c=]
module.wacnet.wireguard_asymmetric_key.peers["gareth"]: Destroying... [id=qMvWamgmt/Lj1GGlJqHrtwhb85GN9UNdIDNFn5AznCI=]
module.wacnet.wireguard_asymmetric_key.peers["ghuynh"]: Destroying... [id=GOE3HMwGwxImSqNn7inR7u9tkxqFZoacCx0uR4BgWnE=]
module.wacnet.wireguard_asymmetric_key.peers["gshevach"]: Destroying... [id=nysu6BL5hw4ApcsTykQmKrSDplFeXSzDOufBZOwv/AQ=]
module.wacnet.wireguard_asymmetric_key.peers["gh-user"]: Destroying... [id=Jgbmri84Jp+idU/qWG/koatv3HuV567z3eYmNW7T1j8=]
module.wacnet.wireguard_asymmetric_key.server: Destroying... [id=5OHMCBnXq3me0h4g+mnGoAw17nAF4Axpn2NIQiyHoFI=]
module.wacnet.wireguard_asymmetric_key.peers["gh-user"]: Destruction complete after 0s
module.wacnet.wireguard_asymmetric_key.peers["mschroeder"]: Destruction complete after 0s
module.wacnet.wireguard_asymmetric_key.peers["gshevach"]: Destruction complete after 0s
module.wacnet.wireguard_asymmetric_key.peers["gh-user"]: Creating...
module.wacnet.wireguard_asymmetric_key.peers["mlibera"]: Destruction complete after 0s
module.wacnet.wireguard_asymmetric_key.peers["gareth"]: Destruction complete after 0s
module.wacnet.wireguard_asymmetric_key.server: Destruction complete after 0s
module.wacnet.wireguard_asymmetric_key.peers["ghuynh"]: Destruction complete after 0s
module.wacnet.wireguard_asymmetric_key.peers["mschroeder"]: Creating...
module.wacnet.wireguard_asymmetric_key.peers["gh-user"]: Creation complete after 0s [id=lCzlt7wXLKLjv1tKI83m0U+kw69CaKwwW7CXRmVfHyY=]
module.wacnet.wireguard_asymmetric_key.peers["mschroeder"]: Creation complete after 0s [id=Zh7K2EzcmroTQy0LoTh8KcGoPDhjlZPE336uRPZa3wA=]
module.wacnet.wireguard_asymmetric_key.peers["gshevach"]: Creating...
module.wacnet.wireguard_asymmetric_key.peers["mlibera"]: Creating...
module.wacnet.wireguard_asymmetric_key.peers["gareth"]: Creating...
module.wacnet.wireguard_asymmetric_key.server: Creating...
module.wacnet.wireguard_asymmetric_key.server: Creation complete after 0s [id=ettLeirIpKKWSv98IQywKvQe96YUNbKg1sWmAAW94Eg=]
module.wacnet.wireguard_asymmetric_key.peers["gareth"]: Creation complete after 0s [id=pXhuiavVqZou6GQzmt1voXU8quBqhtGFhVqAkDTbfhc=]
module.wacnet.wireguard_asymmetric_key.peers["gshevach"]: Creation complete after 0s [id=obxjI2cpceOXBgoaSo9/lw8QA02aNBeAf1STeFQiKA8=]
module.wacnet.wireguard_asymmetric_key.peers["ghuynh"]: Creating...
module.wacnet.wireguard_asymmetric_key.peers["mlibera"]: Creation complete after 0s [id=+dhFlHNdcy63p7oCxNGdiyENuAwq1X2nPuTjCn6WJCc=]
module.wacnet.wireguard_asymmetric_key.peers["ghuynh"]: Creation complete after 0s [id=7ynIkaYgfA1ke1fFujza4NtVBEhLzpLr9CHkUdeYimY=]
module.wacnet.aws_instance.wg_gateway: Creating...
module.wacnet.aws_instance.wg_gateway: Still creating... [10s elapsed]
module.wacnet.aws_instance.wg_gateway: Creation complete after 14s [id=i-04ceaa197c9b104f2]
module.wacnet.aws_eip.wg_eip: Modifying... [id=eipalloc-0949320ac800b43c3]
module.wacnet.aws_eip.wg_eip: Modifications complete after 2s [id=eipalloc-0949320ac800b43c3]
module.wacnet.aws_eip_association.wg_association: Creating...
module.wacnet.aws_eip_association.wg_association: Creation complete after 2s [id=eipassoc-0379deae1a2670f67]
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gareth"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" changed the planned action from
│ Update to DeleteThenCreate.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gareth"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gareth"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .arn: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gareth"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool).
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gareth"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gareth"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_stages: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gareth"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .secret_binary: inconsistent values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gh-user"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" changed the planned action from
│ Update to DeleteThenCreate.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gh-user"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool).
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gh-user"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gh-user"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_stages: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gh-user"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .secret_binary: inconsistent values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gh-user"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gh-user"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .arn: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["ghuynh"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" changed the planned action from
│ Update to DeleteThenCreate.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["ghuynh"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .arn: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["ghuynh"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool).
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["ghuynh"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["ghuynh"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_stages: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["ghuynh"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .secret_binary: inconsistent values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["ghuynh"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mschroeder"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" changed the planned action from
│ Update to DeleteThenCreate.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mschroeder"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_stages: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mschroeder"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .secret_binary: inconsistent values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mschroeder"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mschroeder"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .arn: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mschroeder"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool).
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mschroeder"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mlibera"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" changed the planned action from
│ Update to DeleteThenCreate.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mlibera"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .arn: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mlibera"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool).
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mlibera"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mlibera"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_stages: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mlibera"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .secret_binary: inconsistent values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["mlibera"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gshevach"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" changed the planned action from
│ Update to DeleteThenCreate.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gshevach"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_stages: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gshevach"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .secret_binary: inconsistent values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gshevach"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .version_id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gshevach"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .arn: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gshevach"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool).
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.wacnet.aws_secretsmanager_secret_version.wg_peer_client_config_version["gshevach"]
│ to include new values learned so far during apply, provider
│ "registry.opentofu.org/hashicorp/aws" produced an invalid new value for
│ .id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
Error: Process completed with exit code 1.
---

```


### Sample Terraform Configuration

  Minimal config: a secret whose `secret_string` is a `templatefile`/interpolation of a
  resource that gets `-replace`'d.

  ```hcl
  resource "tls_private_key" "k" {
    algorithm = "ED25519"
  }

  resource "aws_secretsmanager_secret" "s" {
    name                    = "repro-${random_id.r.hex}"
    recovery_window_in_days = 0
  }

  resource "aws_secretsmanager_secret_version" "v" {
    secret_id     = aws_secretsmanager_secret.s.id
    secret_string = tls_private_key.k.private_key_pem
  }

  resource "random_id" "r" { byte_length = 4 }
```

 

### Steps to Reproduce

  1. tofu apply — succeeds.
  2. tofu apply -replace=tls_private_key.k — fails on 6.45.0, succeeds on 6.44.0.

### Debug Logging

<details open>
<summary>Click to expand log output</summary>

Running with `6.45.0`

```console
Plan Generated: 05/14/2026 09:28:04 AM EDT
----------------------------------------

OpenTofu will perform the following actions:

  # aws_secretsmanager_secret.s will be created
  + resource "aws_secretsmanager_secret" "s" {
      + arn                            = (known after apply)
      + force_overwrite_replica_secret = false
      + id                             = (known after apply)
      + name                           = (known after apply)
      + name_prefix                    = (known after apply)
      + policy                         = (known after apply)
      + recovery_window_in_days        = 0
      + region                         = "us-east-1"
      + tags_all                       = (known after apply)

      + replica (known after apply)
    }

  # aws_secretsmanager_secret_version.v will be created
  + resource "aws_secretsmanager_secret_version" "v" {
      + arn                  = (known after apply)
      + has_secret_string_wo = (known after apply)
      + id                   = (known after apply)
      + region               = "us-east-1"
      + secret_id            = (known after apply)
      + secret_string        = (sensitive value)
      + secret_string_wo     = (write-only attribute)
      + version_id           = (known after apply)
      + version_stages       = (known after apply)
    }

  # random_id.r will be created
  + resource "random_id" "r" {
      + b64_std     = (known after apply)
      + b64_url     = (known after apply)
      + byte_length = 4
      + dec         = (known after apply)
      + hex         = (known after apply)
      + id          = (known after apply)
    }

  # tls_private_key.k will be created
  + resource "tls_private_key" "k" {
      + algorithm                     = "ED25519"
      + ecdsa_curve                   = "P224"
      + id                            = (known after apply)
      + private_key_openssh           = (sensitive value)
      + private_key_pem               = (sensitive value)
      + private_key_pem_pkcs8         = (sensitive value)
      + public_key_fingerprint_md5    = (known after apply)
      + public_key_fingerprint_sha256 = (known after apply)
      + public_key_openssh            = (known after apply)
      + public_key_pem                = (known after apply)
      + rsa_bits                      = 2048
    }

Plan: 4 to add, 0 to change, 0 to destroy.

---
apply then tofu apply -replace=tls_private_key.k

  # aws_secretsmanager_secret_version.v will be updated in-place
  ~ resource "aws_secretsmanager_secret_version" "v" {
        id               = "arn:aws:secretsmanager:us-east-1:951625199029:secret:repro-e2fec34c-LpRM9Y|terraform-20260514132921256400000002"
      ~ secret_string    = (sensitive value)
      ~ secret_string_wo = (write-only attribute)
        # (5 unchanged attributes hidden)
    }

  # tls_private_key.k will be replaced, as requested
-/+ resource "tls_private_key" "k" {
      ~ id                            = "16ce83ba52b0456db2586fff6338f715d73ed75c" -> (known after apply)
      ~ private_key_openssh           = (sensitive value)
      ~ private_key_pem               = (sensitive value)
      ~ private_key_pem_pkcs8         = (sensitive value)
      ~ public_key_fingerprint_md5    = "af:25:e5:90:7d:49:a7:82:6b:22:41:6f:a3:82:42:6e" -> (known after apply)
      ~ public_key_fingerprint_sha256 = "SHA256:eWMliHVYSkVqH3sdqp2zWeZrtaiSdAh833q6qKjiiJo" -> (known after apply)
      ~ public_key_openssh            = <<-EOT
            ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKTs95ZBj9oQVSG5mab64aYQKPzgavGJhl3f3ubZG7Ij
        EOT -> (known after apply)
      ~ public_key_pem                = <<-EOT
            -----BEGIN PUBLIC KEY-----
            MCowBQYDK2VwAyEApOz3lkGP2hBVIbmZpvrhphAo/OBq8YmGXd/e5tkbsiM=
            -----END PUBLIC KEY-----
        EOT -> (known after apply)
        # (3 unchanged attributes hidden)
    }

Plan: 1 to add, 1 to change, 1 to destroy.

Do you want to perform these actions?
  OpenTofu will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

tls_private_key.k: Destroying... [id=16ce83ba52b0456db2586fff6338f715d73ed75c]
tls_private_key.k: Destruction complete after 0s
tls_private_key.k: Creating...
tls_private_key.k: Creation complete after 0s [id=cd0e7bf490e39636d16ae454ffb3661c49960b8f]
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_secretsmanager_secret_version.v to include new values learned so far during apply, provider "registry.opentofu.org/hashicorp/aws" changed
│ the planned action from Update to DeleteThenCreate.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_secretsmanager_secret_version.v to include new values learned so far during apply, provider "registry.opentofu.org/hashicorp/aws" produced
│ an invalid new value for .version_stages: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_secretsmanager_secret_version.v to include new values learned so far during apply, provider "registry.opentofu.org/hashicorp/aws" produced
│ an invalid new value for .arn: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_secretsmanager_secret_version.v to include new values learned so far during apply, provider "registry.opentofu.org/hashicorp/aws" produced
│ an invalid new value for .secret_binary: inconsistent values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_secretsmanager_secret_version.v to include new values learned so far during apply, provider "registry.opentofu.org/hashicorp/aws" produced
│ an invalid new value for .version_id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_secretsmanager_secret_version.v to include new values learned so far during apply, provider "registry.opentofu.org/hashicorp/aws" produced
│ an invalid new value for .has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool).
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_secretsmanager_secret_version.v to include new values learned so far during apply, provider "registry.opentofu.org/hashicorp/aws" produced
│ an invalid new value for .id: was known, but now unknown.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.


```
</details>


### GenAI / LLM Assisted Development

n/a

### Important Facts and References


  After upgrading from `hashicorp/aws` 6.44.0 → 6.45.0, `tofu apply` fails with
  "Provider produced inconsistent final plan" on every `aws_secretsmanager_secret_version`
  whose `secret_string` is computed from an upstream resource being replaced in the same
  apply. No other changes to config, OpenTofu, or workflow.

  Our resources use plain `secret_string` only — we do not set `secret_string_wo` anywhere.
  The errors specifically reference the new `has_secret_string_wo` attribute:

  > `has_secret_string_wo: was null, but now cty.UnknownVal(cty.Bool)`

  and report the provider flipping the planned action mid-apply:

  > changed the planned action from Update to DeleteThenCreate

  This appears to be a regression introduced by the 6.45.0 enhancement
  ["Allow switching from `secret_string` to `secret_string_wo` without re-creating
  the resource"](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v6.45.0),
  which is likely the upstream fix for #41635. The new logic seems to engage even when
  `secret_string_wo` is never used, in any apply where `secret_string` resolves from
  `(known after apply)` values.


### Would you like to implement a fix?

No

--- a/terraform/environments/bootstrap/delegate-access/versions.tf
+++ b/terraform/environments/bootstrap/delegate-access/versions.tf
@@ -1,7 +1,7 @@
 terraform {
   required_providers {
     aws = {
-      version = "~> 6.0"
+      version = ">= 6.0, < 6.45.0"
       source  = "hashicorp/aws"
     }
     tls = {

--- a/terraform/environments/bootstrap/member-bootstrap/versions.tf
+++ b/terraform/environments/bootstrap/member-bootstrap/versions.tf
@@ -1,7 +1,7 @@
 terraform {
   required_providers {
     aws = {
-      version = "~> 6.0"
+      version = ">= 6.0, < 6.45.0"
       source  = "hashicorp/aws"
     }
     tls = {

--- a/terraform/environments/bootstrap/secure-baselines/versions.tf
+++ b/terraform/environments/bootstrap/secure-baselines/versions.tf
@@ -1,7 +1,7 @@
 terraform {
   required_providers {
     aws = {
-      version = "~> 6.0"
+      version = ">= 6.0, < 6.45.0"
       source  = "hashicorp/aws"
     }
   }

--- a/terraform/environments/bootstrap/single-sign-on/versions.tf
+++ b/terraform/environments/bootstrap/single-sign-on/versions.tf
@@ -1,7 +1,7 @@
 terraform {
   required_providers {
     aws = {
-      version = "~> 6.0"
+      version = ">= 6.0, < 6.45.0"
       source  = "hashicorp/aws"
     }
   }

--- a/terraform/environments/versions.tf
+++ b/terraform/environments/versions.tf
@@ -1,7 +1,7 @@
 terraform {
   required_providers {
     aws = {
-      version = "~> 6.0"
+      version = ">= 6.0, < 6.45.0"
       source  = "hashicorp/aws"
     }
     github = {

